header-rewrite-filter/header_processor.cc

```diff
@@ -233,47 +238,86 @@ namespace HeaderRewriteFilter {
                     return absl::InvalidArgumentError("too many arguments for set bool");
                 }
             } else if (operation_expression.size() > Utility::SET_BOOL_MIN_NUM_ARGUMENTS) { // if match type does not have an arg
+                std::cerr << "min num args is " << int(Utility::SET_BOOL_MIN_NUM_ARGUMENTS);
+                std::cerr << "num args provided is " << int(operation_expression.size());
                 return absl::InvalidArgumentError("too many arguments for set bool");
             }
 
-            const std::string string_to_compare = std::string(*(start + 4));
+            function_type_ = Utility::StringToFunctionType(dynamic_function);
+            if (function_type_ == Utility::FunctionType::InvalidFunctionType) {
+                return absl::InvalidArgumentError("invalid function type for dynamic value");
+            }
+            const std::tuple<absl::Status, std::string> get_function_argument_result = Utility::GetFunctionArgument(*(start + 1));
+            const absl::Status status = std::get<0>(get_function_argument_result);
+            if (status != absl::OkStatus()) {
+                return status;
+            }
+            function_argument_ = std::get<1>(get_function_argument_result);
+
+            // parse optional argument
+            std::string string_to_compare;
+            if (match_type != Utility::MatchType::Found) {
```

Comment:
nit: move this to the switch case - e.g. imagine you add a new match type which doesn't have an arg and you'd need to update this separately